bin/test_travis.sh

```diff
@@ -106,10 +106,4 @@ export LD_LIBRARY_PATH=$our_install_dir/lib:$LD_LIBRARY_PATH
 echo "Checking whether all header files are installed:"
 python $SOURCE_DIR/symengine/utilities/tests/test_make_install.py $our_install_dir/include/symengine/ $SOURCE_DIR/symengine
 
-# check trailing whitespace:
-if !  egrep " $" -nr --include=\*.{cpp,h,inc}  --exclude-dir=*{teuchos,/build/}* $SOURCE_DIR ; then
-    echo No trailing whitespace;
-else
-    exit -1;
-fi
-# TODO: Add similar grep checks for space after comma,, space after `if`, space between `)` and `{` also
+git diff HEAD^1 --check #test for whitespaces
```

Comment:
It doesn't remove the top commit. Just returns the `diff` introdued at it, i.e. the changes introduced in previous commit.
And `diff --check` is to check for trailing whitespaces.

> --check

    Warn if changes introduce whitespace errors.
    What are considered whitespace errors is controlled by core.whitespace configuration.
    By default, trailing whitespaces (including lines that solely consist of whitespaces) and a space character that is immediately followed by a tab character inside the initial indent of the line are considered whitespace errors.
    Exits with non-zero status if problems are found.